api/app/fireweather_bot/hourly_actuals.py

```diff
@@ -18,16 +18,26 @@
 logger = logging.getLogger(__name__)
 
 
-def _fix_pandas_datetime(panda_datetime):
-    """ We want to be explicit about timezones, pandas gives us a timezone naive date - this is no
+def _fix_datetime(source_date):
+    """ We want to be explicit about timezones, the csv file gives us a timezone naive date - this is no
     good! We know the date we get is in PST, and we know we need to store it in UTC.
+    Furthermore, the csv has discovered another hour. 24. Presumably this means 00 the next day.
     """
-    # go from pandas to python native:
-    python_date = panda_datetime.to_pydatetime()
-    # we want to work in utc:
-    python_date = python_date.replace(tzinfo=timezone.utc)
-    # but alse need to adjust our time, because it's in PST:
+    # build a date using year, month and day.
+    python_date = datetime.strptime(str(source_date)[:-2], '%Y%m%d')
+    # handle the mysterious 24th hour.
+    if str(source_date)[-2:] == '24':
+        # if it's 24, just push to hour 00 on the next day.
+        python_date = python_date + timedelta(days=1)
+    else:
+        # anything else, we trust.
+        python_date = python_date + timedelta(hours=int(str(source_date)[-2:]))
+    # but alse need to adjust our time, because it's in PST. PST_UTC_OFFSET is -8, so we have
+    # to subtract it, to add 8 hours to our current date.
+    # e.g. 12h00 PST, becomes 20h00 UTC
     python_date = python_date - timedelta(hours=app.time_utils.PST_UTC_OFFSET)
+    # we've add the offset, so set the timezone to utc:
+    python_date = python_date.replace(tzinfo=timezone.utc)
```

Comment:
@Kyubinhan I've added some comments and changed the order of operations so it's a bit more clear what's happening.